File: Module2/src/_07_abstract_interface/bt/resizeable/Rectangle.java
package _07_abstract_interface.bt.resizeable;

public class Rectangle extends Shape implements Resizeable {
    private double width;
    private double heigth;

    public Rectangle(String color, double width, double heigth) {
        super(color);
        this.width = width;
        this.heigth = heigth;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getHeigth() {
        return heigth;
    }

    public void setHeigth(double heigth) {
        this.heigth = heigth;
    }



    @Override
    public double resize(double percent) {
        return this.getArea()+percent/100*this.getArea();
    }

    @Override
    public double getArea() {
        return this.heigth*this.width;
    }
}

File: Module5/1-Typescript/8-Inheritance/Rectangle.ts
import { Shape } from "./Shape";
export class Rectangle extends Shape {

    constructor(theX: number, theY:number,private _length: number,private _width: number) {
        super(theX,theY);
    }
    public get width(): number {
        return this._width;
    }
    public set width(value: number) {
        this._width = value;
    }
    public get length(): number {
        return this._length;
    }
    public set length(value: number) {
        this._length = value;
    }

    getInfor():string{
        return super.getInfor() + `,length=${this.length}, width=${this.length}`;
    }
}